packages/ifpack2/src/Ifpack2_LocalSparseTriangularSolver_def.hpp

```diff
@@ -398,6 +398,34 @@ compute ()
   if (Teuchos::nonnull (htsImpl_))
     htsImpl_->compute (*A_crs_, out_);
 
+  // non-completed code to reverse the storage order of the local matrix
+  typedef  Tpetra::CrsMatrix<scalar_type,local_ordinal_type,global_ordinal_type,node_type> crs_matrix_type;
+  typename crs_matrix_type::local_matrix_type A = A_crs_->getLocalMatrix();
+  typename crs_matrix_type::local_matrix_type::row_map_type ptr = A.graph.row_map;
+  typename crs_matrix_type::local_matrix_type::index_type   ind = A.graph.entries;
+  typename crs_matrix_type::local_matrix_type::values_type  val = A.values;
+
+  const local_ordinal_type numRows = A.numRows ();
+
+  typename crs_matrix_type::local_matrix_type::index_type::non_const_type newind ("ind", ind.dimension_0 ());
+  typename crs_matrix_type::local_matrix_type::values_type newval ("val", val.dimension_0 ());
+
+  local_ordinal_type lclNumRows = A.numRows ();
+  local_ordinal_type rowStart = 0; 
+  for (local_ordinal_type lclRow = 0; lclRow < lclNumRows; ++lclRow) {
+     auto A_r = A.row (lclNumRows-lclRow-1);
+     const local_ordinal_type numEnt = A_r.length;
+     for (local_ordinal_type k = 0; k < numEnt; ++k) {
+        newval(rowStart + k) = A_r.values(numEnt - k - 1);
+        newcol(rowStart + k) = A_r.colidx(numEnt - k - 1);
+     }
+     rowStart = rowStart + numEnt;
```

Comment:
@sthomas61 Do you actually need to change the entries of `ptr`?  If so, then you need to make a new allocation.